File: ShipTest/src/ShipTest.cpp
#include <iostream>
#include "ship.h"

void
output(Ship ship);
using namespace std;

int
main()
{
  Ship test;
  float x, y; //general x and y coordinate variable, used for input
  float ang; //used for angle input
  int vel;//input
  //////////////////////////////////////////////////////////////////////////////
  cout << "Data is expressed as Location (x,y), Velocity <x,y>, Angle" << endl
       << "Press [ENTER] to continue";
  cin.get();
  output(test);

  /////////////////////////////////////////////////////////////////////////////////

  cout << "Set Location (x y): ";
  cin >> x >> y;
  test.setLocation(x, y);
  cout << "Set Velocity (x y): ";
  cin >> x >> y;
  test.setVelocity(x, y);

  cout << "Set Angle (0 to 359): ";
  cin >> ang;
  test.setAngle(ang);

  output(test);

  ////////////////////////////////////////////////////////////////////////////////

  cout << "Input degrees to turn (positive turns left): ";
  cin >> ang;
  if (ang > 0)
    for (int i = 0; i < ang; i++)
      test.turnLeft();
  else if (ang < 0)
    for (int i = 0; i < abs(ang); i++)
      test.turnRight();

  test.updateLocation();
  output(test);

  cout << "Input the velocity to go forward (int): ";
  cin >> vel;
  test.applyThrust(vel);
  test.updateLocation();
  output(test);
}

/////////////////////////////////////////////////////////////////////////////////

void
output(Ship ship)
{
  MyVect loc = ship.getLocation();
  MyVect vel = ship.getVelocity();
  float ang = ship.getAngle();
  cout << "(" << loc.x << "," << loc.y << "); <"
      << vel.x << "," << vel.y << ">; "
      << ang << endl;
}

File: ShipTestPy/src/ShipTest.py
from Ship import Ship, output, listToFloat

test = Ship()#these parenthesis necessary, they allow the class
                #to instantiate

print("Data is expressed as Location (x,y), Velocity <x,y>, Angle")
input("Press [ENTER] to continue")

loc = input("Set Location (x y): ").split()
listToFloat(loc)
test.setLocation(loc)

vel = input("Set Velocity (x y): ").split()
listToFloat(vel)
test.setVelocity(vel)

ang = int(input("Set Angle (0 to 359): "))
test.setAngle(ang)

output(test)
#####################################################################
while True:
    ang = int(input("Enter degress to turn (+left): "))
    if ang > 0:
        for i in range(ang):
            test.turnLeft()
    elif ang < 0:
        for i in range(abs(ang)):
            test.turnRight()
    test.updateLocation()
    output(test)
    
    angVel = float(input("Input the velocity to go forward (float): "))
    test.applyThrust(angVel)
    test.updateLocation()
    output(test)